range: compile warning, failed tests

compile warning:
```
/home/markus/Projekte/Elektra/current/src/plugins/range/range.c: In function ‘strToValue’:
/home/markus/Projekte/Elektra/current/src/plugins/range/range.c:97:1: note: the ABI of passing union with long double has changed in GCC 4.4
 {
 ^
/home/markus/Projekte/Elektra/current/src/plugins/range/range.c:115:51: warning: comparing floating point with == or != is unsafe [-Wfloat-equal]
   if (errno == ERANGE || (errno != 0 && v.Value.f == 0))
                                                   ^
```

failed test:
```
 88/126 Test  #94: testmod_range ....................***Failed    0.00 sec
RANGE     TESTS
==================

/home/markus/Projekte/Elektra/current/src/plugins/range/testmod_range.c:61: error in testFloat: failed
/home/markus/Projekte/Elektra/current/src/plugins/range/testmod_range.c:61: error in testFloat: failed

testmod_range RESULTS: 204 test(s) done. 2 error(s).
```

--- a/src/plugins/range/CMakeLists.txt
+++ b/src/plugins/range/CMakeLists.txt
@@ -1,5 +1,12 @@
 include (LibAddMacros)
 
+if (DEPENDENCY_PHASE)
+    set_source_files_properties(range.c PROPERTIES COMPILE_FLAGS -Wno-psabi)
+endif ()
+if(UNIX)
+    link_libraries(m)
+endif(UNIX)
+
 add_plugin (range
 	SOURCES
 		range.h

--- a/src/plugins/range/CMakeLists.txt
+++ b/src/plugins/range/CMakeLists.txt
@@ -3,13 +3,12 @@ include (LibAddMacros)
 if (DEPENDENCY_PHASE)
     set_source_files_properties(range.c PROPERTIES COMPILE_FLAGS -Wno-psabi)
 endif ()
-if(UNIX)
-    link_libraries(m)
-endif(UNIX)
 
 add_plugin (range
 	SOURCES
 		range.h
 		range.c
+	LINK_LIBRARIES
+		"-lm"
 	ADD_TEST
 	)

--- a/src/plugins/range/README.md
+++ b/src/plugins/range/README.md
@@ -17,7 +17,7 @@ The range plugin checks if a Key value is within a given range.
 
 The plugin checks every Key in the Keyset for the Metakey `check/range` which contains either a single range with the syntax `[-]min-[-]max`, or a list of ranges or values separated by `,` and tests if the Key value is within the range(s).
 
-`check/type` or `type` can be used to specify the data type. If not specified otherwise the default value is `INT`
+`check/type` can be used to specify the datatype. If not specified otherwise the default value is `long long`
 
 Possible values:
 

--- a/src/plugins/range/range.c
+++ b/src/plugins/range/range.c
@@ -12,6 +12,7 @@
 #include <errno.h>
 #include <kdberrors.h>
 #include <kdbhelper.h>
+#include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -112,7 +113,7 @@ RangeValue strToValue (const char ** ptr, RangeType type)
 		break;
 	case FLOAT:
 		v.Value.f = strtold (*ptr, &endPtr);
-		if (errno == ERANGE || (errno != 0 && v.Value.f == 0))
+		if (errno == ERANGE || (errno != 0 && fpclassify (v.Value.f) == FP_ZERO))
 		{
 			v.type = NA;
 		}

--- a/src/plugins/range/range.c
+++ b/src/plugins/range/range.c
@@ -401,7 +401,7 @@ static RangeType getType (const Key * key)
 		return type;
 }
 
-int validateKey (Key * key, Key * parentKey)
+static int validateKey (Key * key, Key * parentKey)
 {
 	const Key * rangeMeta = keyGetMeta (key, "check/range");
 	const char * rangeString = keyString (rangeMeta);

--- a/src/plugins/range/range.c
+++ b/src/plugins/range/range.c
@@ -390,14 +390,11 @@ static RangeType stringToType (const Key * typeMeta)
 
 static RangeType getType (const Key * key)
 {
-	const Key * typeMeta = keyGetMeta (key, "type");
+	const Key * typeMeta = keyGetMeta (key, "check/type");
 	RangeType type = NA;
 
 	type = stringToType (typeMeta);
 
-	if (type == NA) typeMeta = keyGetMeta (key, "check/type");
-
-	type = stringToType (typeMeta);
 	if (type == NA)
 		return INT;
 	else

--- a/src/plugins/range/testmod_range.c
+++ b/src/plugins/range/testmod_range.c
@@ -34,8 +34,8 @@ void testInt (const char * value, int ret, const char * rangeString)
 void testUInt (const char * value, int ret, const char * rangeString)
 {
 	Key * parentKey = keyNew ("user/tests/range", KEY_VALUE, "", KEY_END);
-	KeySet * ks = ksNew (10, keyNew ("user/tests/range/key", KEY_VALUE, value, KEY_META, "check/range", rangeString, KEY_META, "type",
-					 "unsigned long", KEY_END),
+	KeySet * ks = ksNew (10, keyNew ("user/tests/range/key", KEY_VALUE, value, KEY_META, "check/range", rangeString, KEY_META,
+					 "check/type", "unsigned long", KEY_END),
 			     KS_END);
 	KeySet * conf = ksNew (0, KS_END);
 	PLUGIN_OPEN ("range");

--- a/src/plugins/range/testmod_range.c
+++ b/src/plugins/range/testmod_range.c
@@ -8,6 +8,7 @@
  */
 
 #include <kdbconfig.h>
+#include <locale.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -105,6 +106,9 @@ int main (int argc, char ** argv)
 
 	init (argc, argv);
 
+	const char * old_locale = setlocale (LC_ALL, NULL);
+	setlocale (LC_ALL, "C");
+
 	testInt ("5", 1, "1-10");
 	testInt ("10", 1, "1-10");
 	testInt ("1", 1, "1-10");
@@ -174,6 +178,7 @@ int main (int argc, char ** argv)
 	testChar ("g", -1, "a-f");
 	testChar ("c", 1, "a-f");
 
+	setlocale (LC_ALL, old_locale);
 	printf ("\ntestmod_range RESULTS: %d test(s) done. %d error(s).\n", nbTest, nbError);
 
 	return nbError;

--- a/tests/valgrind.suppression
+++ b/tests/valgrind.suppression
@@ -270,3 +270,10 @@
    fun:_ZN61kdbrestServicesConvertengineTest_FindSuitablePluginCheck_Test8TestBodyEv
    ...
 }
+{
+   testmod_range setlocale memleak
+   Memcheck:Addr1
+   ...
+   fun:setlocale
+   fun:main
+}

